client/src/http.rs

```diff
@@ -10,11 +10,20 @@ pub use http::{Method, Response, StatusCode};
 ///
 /// The order of builder methods invocation is not strict. There is no guarantee that builder user calls
 /// all methods. Only [`RequestBuilder::new`] is the required one.
-pub trait RequestBuilder {
-    /// Entrypoint - create a new builder with specified method and URL.
+pub trait RequestBuilder
+where
+    Self: Sized,
+{
     #[must_use]
+    /// Entrypoint - create a new builder with specified method and URL.
     fn new(method: Method, url: impl AsRef<str>) -> Self;
 
+    /// Entrypoint - create a new builder with specified method and URL.
+    ///
+    /// # Errors
+    /// If either the `method` or `url` are invalid.
+    fn try_new(method: Method, url: impl AsRef<str> + core::fmt::Display) -> eyre::Result<Self>;
```

Comment:
What is the point of `try_new` method? How should the `RequestBuilder` user decide whether to use `new` or `try_new` method to create a builder?

Reply:
Also, `try_new` method should have the `#[must_use]` attr as well, I guess.

Reply:
I see. LGTM, but I'd persist on adding `#[deprecated]` for `new` method and `#[must_use]` for `try_new`.

Reply:
Certainly.